src/XCCDF_POLICY/xccdf_policy.c

```diff
@@ -572,12 +573,26 @@ _xccdf_policy_rule_get_applicable_check(struct xccdf_policy *policy, struct xccd
 			// If the refined selector does not match, checks without selector shall be used.
 			candidate_it = xccdf_rule_get_checks_filtered(rule, NULL);
 		}
+
+		bool print_warning = false;
+		char *warning_check_system;
 		// Check Processing Algorithm -- Check.System
 		while (xccdf_check_iterator_has_more(candidate_it)) {
 			struct xccdf_check *check = xccdf_check_iterator_next(candidate_it);
-			if (_xccdf_policy_is_engine_registered(policy, (char *) xccdf_check_get_system(check)))
+			if (_xccdf_policy_is_engine_registered(policy, (char *) xccdf_check_get_system(check))) {
 				result = check;
+			} else {
+				print_warning = true;
+				warning_check_system = check->system;
+			}
 		}
+
+		// Only print a warning if we didn't select a check but could've otherwise.
+		if (print_warning && result == NULL)
+			printf("WARNING: Skipping rule that requires an unregistered check system "
+			       "or incorrect content reference to evaluate. "
+			       "Please consider providing a valid SCAP/OVAL instead of %s\n",
+				warning_check_system);
```

Comment:
Missed @mpreisler and @cipherboy's original ping. What's the nuance of "unregistered check system" vs "unsupported check system"?

Reply:
OpenSCAP may support a check system - say SCE - but the plugin may not be installed at the time.

Reply:
Given the current texts, how would a user identify if the rule doesn't have a check component at all versus if it's just something that hasn't been implemented yet?

If a user opens a support ticket, how would someone help troubleshoot the difference? Would throwing a unique error/warning for each condition help decrease support burdens?

Reply:
@shawndwells A [rule](https://github.com/cipherboy/openscap/blob/4fb933b860305dfefd6e1ab8c053b75403edb671/src/XCCDF_POLICY/xccdf_policy.c#L568) will have multiple [checks](https://github.com/cipherboy/openscap/blob/4fb933b860305dfefd6e1ab8c053b75403edb671/src/XCCDF_POLICY/xccdf_policy.c#L570). If [a check](https://github.com/cipherboy/openscap/blob/4fb933b860305dfefd6e1ab8c053b75403edb671/src/XCCDF_POLICY/xccdf_policy.c#L570) would've been selected if [it was registered](https://github.com/cipherboy/openscap/blob/4fb933b860305dfefd6e1ab8c053b75403edb671/src/XCCDF_POLICY/xccdf_policy.c#L582), it will [print a warning](https://github.com/cipherboy/openscap/blob/4fb933b860305dfefd6e1ab8c053b75403edb671/src/XCCDF_POLICY/xccdf_policy.c#L585) if it wasn't able to select an [appropriate check](https://github.com/cipherboy/openscap/blob/4fb933b860305dfefd6e1ab8c053b75403edb671/src/XCCDF_POLICY/xccdf_policy.c#L591) because either (the check system is unregistered / proprietary / invalid / etc.) or (it was a supported check system (e.g., OVAL), but the reference to the specific check was invalid (`check-content-ref` pointed to a missing file / invalid path / etc)).


Thus, this message won't print if a rule doesn't have a check so that scenario won't happen. More likely, a warning (which won't halt execution) will be printed:
 - If the OVAL referenced check is wrong (in which case the warning will print the OVAL reference URL and thus can be identified that way). 
 - Or, if the system used a proprietary check system (in which case the URL will not be OVAL and will be something else, which will also be printed) and no valid check systems (OVALs) were supplied as alternatives. 

I'd have to check, but to my knowledge, a warning is not printed when a rule has no checks; perhaps this should be added in a separate PR.